engine/resources/mount.go

```diff
@@ -0,0 +1,712 @@
+	return false, nil
+}
+
+// mountCheckApply checks if the defined resource is mounted, and mounts or
+// unmounts it according to the defined state.
+func (obj *MountRes) mountCheckApply(apply bool) (bool, error) {
+	exists, err := mountExists(procPath, obj.mount)
+	if err != nil {
+		return false, errwrap.Wrapf(err, "error checking if mount exists")
+	}
+
+	// if everything is as it should be, we're done
+	if (exists && obj.State == "exists") || (!exists && obj.State == "absent") {
+		return true, nil
+	}
+
+	if !apply {
+		return false, nil
+	}
+
+	obj.init.Logf("mountCheckApply(%t)", apply)
+
+	if obj.State == "exists" {
+		// reload mounts from /etc/fstab
+		if err := mountReload(); err != nil {
+			return false, errwrap.Wrapf(err, "error reloading /etc/fstab")
+		}
+		return false, nil // we're done
+	}
+	// unmount the device
+	if err := unix.Unmount(obj.Name(), 0); err != nil {
+		return false, errwrap.Wrapf(err, "error unmounting %s", obj.Name())
+	}
+	return false, nil
+}
+
+// CheckApply is run to check the state and, if apply is true, to apply the
+// necessary changes to reach the desired state. This is run before Watch and
+// again if Watch finds a change occurring to the state.
+func (obj *MountRes) CheckApply(apply bool) (checkOK bool, err error) {
+	checkOK = true
+
+	if c, err := obj.fstabCheckApply(apply); err != nil {
+		return false, err
+	} else if !c {
+		checkOK = false
+	}
+
+	if c, err := obj.mountCheckApply(apply); err != nil {
+		return false, err
+	} else if !c {
+		checkOK = false
+	}
+
+	return checkOK, nil
+}
+
+// MountUID is a unique resource identifier.
+type MountUID struct {
+	engine.BaseUID
+	name string
+}
+
+// IFF aka if and only if they are equivalent, return true. If not, false.
+func (obj *MountUID) IFF(uid engine.ResUID) bool {
+	res, ok := uid.(*MountUID)
+	if !ok {
+		return false
+	}
+	return obj.name == res.name
+}
+
+// UIDs includes all params to make a unique identification of this object.
+// Most resources only return one although some resources can return multiple.
+func (obj *MountRes) UIDs() []engine.ResUID {
+	x := &MountUID{
+		BaseUID: engine.BaseUID{Name: obj.Name(), Kind: obj.Kind()},
+		name:    obj.Name(),
+	}
+	return []engine.ResUID{x}
+}
+
+// Cmp compares two resources and return if they are equivalent.
+func (obj *MountRes) Cmp(r engine.Res) error {
+	// we can only compare MountRes to others of the same resource kind
+	res, ok := r.(*MountRes)
+	if !ok {
+		return fmt.Errorf("not a %s", obj.Kind())
+	}
+	if obj.State != res.State {
+		return fmt.Errorf("the State differs")
+	}
+	if obj.Type != res.Type {
+		return fmt.Errorf("the Type differs")
+	}
+	if !strMapEq(obj.MntOps, res.MntOps) {
+		return fmt.Errorf("the MntOps differ")
+	}
+	if obj.Freq != res.Freq {
+		return fmt.Errorf("the Type differs")
+	}
+	if obj.PassNo != res.PassNo {
+		return fmt.Errorf("the PassNo differs")
+	}
+
+	return nil
+}
+
+// Close is run by the engine to clean up after the resource is done.
```

Comment:
Can you reorder the method order in this file to match what is done in the other resources please?